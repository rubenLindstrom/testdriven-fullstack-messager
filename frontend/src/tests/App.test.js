import React from "react";
import MessageApp from "../App";

import mockAxios from "../__mocks__/axios";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  beforeEach(() => {
    mockAxios.post.mockImplementation(() => Promise.resolve({ data: [] }));
  });

  afterEach(() => mockAxios.post.mockClear());

  it("renders without crashing", () => {
    const component = mount(<MessageApp />);
    expect(component).toMatchSnapshot();
  });

  it("has submit message list", () => {
    const component = mount(<MessageApp />);
    expect(component.exists("ul#message_list")).toBe(true);
  });

  it("posts data and clears message box on submit success", () => {
    const component = mount(<MessageApp />);
    const childComponent = component.find("MessageForm");
    component.find("textarea#message_box").simulate("change", {
      target: { value: "Hello" },
    });
    component.find("form").simulate("submit");

    expect(mockAxios.post).toHaveBeenCalledWith(
      "http://localhost:3001/message",
      { content: "Hello" }
    );

    expect(childComponent.state("currentMessage")).toEqual("");
  });
});
