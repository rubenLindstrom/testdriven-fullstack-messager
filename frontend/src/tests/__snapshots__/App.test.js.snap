// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `ReactWrapper {}`;

exports[`MessageApp renders without crashing 1`] = `ReactWrapper {}`;
